vortex-api/src/main/java/mil/army/usace/hec/vortex/io/NetcdfDataWriter.java

```diff
@@ -1,19 +1,25 @@
 package mil.army.usace.hec.vortex.io;
 
 import mil.army.usace.hec.vortex.VortexGrid;
+import mil.army.usace.hec.vortex.VortexProperty;
 import ucar.nc2.Attribute;
 import ucar.nc2.write.Nc4Chunking;
 import ucar.nc2.write.Nc4ChunkingStrategy;
 import ucar.nc2.write.NetcdfFileFormat;
 import ucar.nc2.write.NetcdfFormatWriter;
 
 import java.beans.PropertyChangeListener;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
-import java.util.stream.Collectors;
+import java.util.Map;
 
 public class NetcdfDataWriter extends DataWriter {
     private final List<VortexGrid> vortexGridList;
-    private final boolean isOverwrite;
+
+    private static final Map<String, Boolean> fileFirstWriteCompleted = Collections.synchronizedMap(new HashMap<>());
```

Comment:
This `Map` makes sure that only the first `ImportableUnit` may overwrite (if specified to overwrite by the user) the destination file, and the rest would append data.